quinn-proto/src/connection/mod.rs

```diff
@@ -520,6 +628,8 @@ where
                     min_datagram_size,
                 );
                 coalesce = false;
+                // We don't want to send 2 close packets
+                close = false;
```

Comment:
I suspect this is the right place to unset `self.close`, as well. Preexisting issue, though, no need to address it in this PR.

Reply:
Right, let's address this later. Not sure if it's sufficient here, because it could get lost on some early returns. We might rather need to keep `self.close` set around until it is really sent.